config/cht/nools-extras.js

```diff
@@ -0,0 +1,342 @@
+  if (isPregnancyFollowUpForm(r)) {
+    if (r.fields && r.fields.lmp_date_8601) {
+      return getDateMS(r.fields.lmp_date_8601);
+    }
+  }
+  return null;
+};
+
+
+
+function getSubsequentPregnancies(c, report) {
+  return c.reports.filter(function (r) {
+    return isPregnancyForm(r) && r.reported_date > report.reported_date;
+  });
+}
+
+function getSubsequentPregnancyFollowUps(c, r) {
+  const subsequentVisits = c.reports.filter(function (v) {
+    let lmpDate = getLMPDateFromPregnancy(r);
+    if (lmpDate === null) { //LMP Date is not available use reported date
+      lmpDate = r.reported_date;
+    }
+
+    return isPregnancyFollowUpForm(v) &&
+      v.reported_date > r.reported_date &&
+      v.reported_date < addDays(lmpDate, MAX_DAYS_IN_PREGNANCY);
+  });
+  return subsequentVisits;
+}
+
+function getSubsequentDeliveries(c, report, withinLastXDays) {
+  return c.reports.filter(function (r) {
+    return (r.form === 'delivery') &&
+      r.reported_date > report.reported_date &&
+      (!withinLastXDays || report.reported_date >= (today - withinLastXDays * MS_IN_DAY));
+  });
+}
+
+function getMostRecentLMPDateForPregnancy(c, r) {
+  let mostRecentLMP = getLMPDateFromPregnancy(r);
+  let mostRecentReportDate = r.reported_date;
+  getSubsequentPregnancyFollowUps(c, r).forEach(function (v) {
+    const lmpFromPregnancyFollowUp = getLMPDateFromPregnancyFollowUp(v);
+    if (v.reported_date > mostRecentReportDate && lmpFromPregnancyFollowUp !== '' && lmpFromPregnancyFollowUp !== mostRecentLMP) {
+      mostRecentReportDate = v.reported_date;
+      mostRecentLMP = lmpFromPregnancyFollowUp;
+    }
+  });
+  return mostRecentLMP;
+}
+
+
+function isPregnancyTerminatedByAbortion(c, r) {
+  const followUps = getSubsequentPregnancyFollowUps(c, r);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup &&
+    latestFollowup.fields.pregnancy_summary &&
+    latestFollowup.fields.pregnancy_summary.visit_option === 'abortion';
+}
+
+function isPregnancyTerminatedByMiscarriage(c, r) {
+  const followUps = getSubsequentPregnancyFollowUps(c, r);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  return latestFollowup &&
+    latestFollowup.fields.pregnancy_summary &&
+    latestFollowup.fields.pregnancy_summary.visit_option === 'miscarriage';
+}
+
+function isActivePregnancy(c, r) {
+  //console.log('check active pregnancy');
+  if (!isPregnancyForm(r)) return false;
+  const lmpDate = getMostRecentLMPDateForPregnancy(c, r);
+  //console.log('c1', lmpDate > today - 294 * MS_IN_DAY, lmpDate, today);
+  //console.log('c2', !getSubsequentDeliveries(c, r, 6 * 7).length);
+  //console.log('c3', !getSubsequentPregnancies(c, r).length);
+  //console.log('c4', !isPregnancyTerminatedByAbortion(c, r));
+  //console.log('c5', isPregnancyTerminatedByMiscarriage(c, r));
+  return lmpDate > today - 294 * MS_IN_DAY && //Pregnancy registration in the past 9 months
```

Comment:
Can you replace all occurrences of `294` with the constant that you have for it? `MAX_DAYS_IN_PREGNANCY`?

Reply:
Here, `MAX_DAYS_IN_PREGNANCY = 44 weeks = 308 days`
There are some sections in the requirement where it is specified to keep as 294 days (42 weeks), so I have kept it as 292 days.
I think the period of 42 weeks is too short for certain other scenarios such as looking for home visit reports for a certain pregnancy, so I have used 44 weeks in such places.
I think some more discussion will be needed with the team to decide what interval to keep in which scenario.

Reply:
Okay interesting. Maybe make a new const for 294 then?

Reply:
We decided to make it 42 weeks (294 days) everywhere except one place where we need 46 weeks (to show delivery form for 280 days + 6 weeks from the date of registration when there is no LMP information). So MAX_DAYS_IN_PREGNANCY is now set as 42 weeks.